lintly/parsers.py

```diff
@@ -64,13 +64,22 @@ def parse_violations(self, output):
             if not match:
                 continue
 
-            path = self._normalize_path(match.group('path'))
+            def get_from_match(key, not_found_value=None):
```

Comment:
In this case I think it's actually less readable to define this function inside a `for` loop. I think this sort of helper function is perfectly acceptable as a method inside the class (so `LineRegexParser.get_from_match()` or `LineRegexParser._get_from_match()` or whichever you prefer).